java/code/src/com/suse/manager/webui/services/SaltStateGeneratorService.java

```diff
@@ -384,9 +384,17 @@ private String getImagePillarFileName(OSImageInspectSlsResult.Bundle bundle) {
      * @param minion the minion server
      */
     public void removePillar(MinionServer minion) {
-        LOG.debug("Removing pillar file for minion: " + minion.getMinionId());
+        removePillar(minion.getMinionId());
+    }
+
+    /**
+     * Remove the corresponding pillar data if the server is a minion.
```

Comment:
Nitpick: We are not checking here if the server is minion, rather it is already assumed so maybe update the method docs accordingly.